### PR TITLE
add wss client support

### PR DESCRIPTION
**Pull request by adam429**
Add wss client support for ethereum.rb

Support node via wss connection "wss://xxxx"

Using the pipeline of wscat.

### CODE REVIEW

**Comment by q9f** [comment]
Hi @adam429 - I'd be happy to accept this contribution to the new `eth` gem at https://github.com/q9f/eth.rb

File: lib/ethereum/client.rb [hunk 1]
```diff
@@ -28,6 +28,7 @@ def initialize(log = false)
     def self.create(host_or_ipcpath, log = false)
       return IpcClient.new(host_or_ipcpath, log) if host_or_ipcpath.end_with? '.ipc'
       return HttpClient.new(host_or_ipcpath, log) if host_or_ipcpath.start_with? 'http'
+      return WssClient.new(host_or_ipcpath, log) if host_or_ipcpath.start_with? 'wss'
```
Comment (q9f) [inline lib/ethereum/client.rb:31]: It should also support insecure websockets.

```suggestion
      return WsClient.new(host_or_ipcpath, log) if host_or_ipcpath.start_with? 'ws'
```

File: lib/ethereum/wss_client.rb [hunk 2]
```diff
@@ -0,0 +1,149 @@
+require 'open3'
+require 'json'
+
+module Ethereum
+  class WssConnection
+    attr_accessor :host,:ws_in,:ws_out,:ws_err,:wait_thr,:index
```
Comment (q9f) [inline lib/ethereum/wss_client.rb:6]: also should have an ssl attribute.

```suggestion
    attr_accessor :host, :ws_in, :ws_out, :ws_err, :wait_thr, :ssl, :index
```

**Comment by kurotaky** [comment]
@adam429 
Thanks for the PR. Please migrate as we will maintain the [eth.rb](https://github.com/q9f/eth.rb/) library here. ;)

**Pull request closed without merging**